SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -22,18 +90,13 @@ def display_package(p):
             'changelog': changelog, 'url': p.url, 'size': p.size, 'license': p.license}
 
 
-def run_command(command):
-    process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    stdout, stderr = process.communicate()
-    result = {'exit': process.returncode, 'stdout': stdout, 'stderr': stderr, 'command': command}
-    _LOGGER.info(result)
-    return result
-
-
+@operationlock()
 def check_update(session, args):
     yum_instance = yum.YumBase()
     yum_instance.preconf.debuglevel = 0
     yum_instance.preconf.plugins = True
+    yum_instance.repos.disableRepo('*')
```

Comment:
Just checking: this only applies to yum_instance, not to the repo files themselves?

Reply:
Just to the instance. ;)